### PR TITLE
2D threshold

### PR DESCRIPTION
**Pull request by JorgeGtz**
**Describe your changes**
This PR contains the function to perform color segmentation using a straight line in a 2D plane formed by two image channels.
This function is linked to the visualization function in PR #810. 

**Type of update**
Is this a:
* New feature or feature enhancement

**Associated issues**

**Additional context**

### CODE REVIEW

**Review by HaleySchuhl — DISMISSED** [review]
Looks great so far. I think I recall some discussion about refactoring the name. Did we still want to try and reduce some of the redundancy?

**Comment by nfahlgren** [comment]
Yeah, I think we discussed removing "threshold" from the function name since it's in the threshold submodule, but don't think we came to a firm decision about the function name

**Review by HaleySchuhl — APPROVED** [review]

File: docs/threshold_threshold_2_channels.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+## Threshold Two Channels
+
+Creates a binary image from an RGB image based on the pixels values in two channels.
+The x and y channels define a 2D plane and the two input points define a straight line.
+Pixels in the plane above and below the straight line are assigned two different values.
+
+**plantcv.threshold.threshold_2_channels**(*rgb_img, x_channel, y_channel, points, above=True, max_value=255*)
```
Comment (nfahlgren) [inline docs/threshold_threshold_2_channels.md:7]:
```suggestion
**plantcv.threshold.dual_channels**(*rgb_img, x_channel, y_channel, points, above=True, max_value=255*)
```

File: docs/updating.md [hunk 2]
```diff
@@ -890,10 +890,15 @@ pages for more details on the input and output variable types.
 * pre v3.9: NA
 * post v3.9: filtered_img = **plantcv.stdev_filter**(*img, kszie, borders="nearest"*)
 
+#### plantcv.threshold.threshold_2_channels
+
+* pre v4.0: NA
+* post v4.0: bin_img **plantcv.threshold.threshold_2_channels**(*rgb_img, x_channel, y_channel, points, above=True, max_value=255*)
```
Comment (nfahlgren) [inline docs/updating.md:896]:
```suggestion
* post v4.0: bin_img **plantcv.threshold.dual_channels**(*rgb_img, x_channel, y_channel, points, above=True, max_value=255*)
```

File: docs/threshold_threshold_2_channels.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+## Threshold Two Channels
```
Comment (nfahlgren) [inline docs/threshold_threshold_2_channels.md:1]: Rename file to `threshold_dual_channels.md`

File: mkdocs.yml [hunk 4]
```diff
@@ -151,6 +151,7 @@ nav:
         - 'Mean Adaptive Threshold': mean_threshold.md
         - 'Otsu Auto Threshold': otsu_threshold.md
         - 'Saturation Threshold': saturation_threshold.md
+        - 'Threshold Two Channels': threshold_threshold_2_channels.md
```
Comment (nfahlgren) [inline mkdocs.yml:154]: Rename file to `threshold_dual_channels.md`

**Review by nfahlgren — CHANGES_REQUESTED** [review]

**Review by nfahlgren — APPROVED** [review]

**Pull request merged**